### PR TITLE
✨ ROSA: Reconcile ROSAControlPlane version

### PR DESCRIPTION
**Pull request by muraee**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Validate and reconcile `.spec.version` to trigger an upgrade if version changes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Reconcile ROSAControlPlane version to trigger upgrades
```

### CODE REVIEW

File: controlplane/rosa/api/v1beta2/rosacontrolplane_types.go [hunk 1]
```diff
@@ -48,8 +48,8 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// The AWS Region the cluster lives in.
 	Region *string `json:"region"`
 
-	// Openshift version, for example "openshift-v4.14.5".
-	Version *string `json:"version"`
+	// Openshift version, for example "4.14.5".
+	Version string `json:"version"`
```
Comment (vincepri) [inline controlplane/rosa/api/v1beta2/rosacontrolplane_types.go:52]: We might want to add some validation here? We could check semver?

File: controlplane/rosa/controllers/rosacontrolplane_controller.go [hunk 2]
```diff
@@ -394,6 +407,41 @@ func (r *ROSAControlPlaneReconciler) reconcileDelete(ctx context.Context, rosaSc
 	return ctrl.Result{RequeueAfter: time.Second * 60}, nil
 }
 
+func (r *ROSAControlPlaneReconciler) reconcileClusterVersion(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope, rosaClient *rosa.RosaClient, cluster *cmv1.Cluster) error {
+	version := rosaScope.ControlPlane.Spec.Version
+	if version == cluster.Version().RawID() {
```
Comment (vincepri) [inline controlplane/rosa/controllers/rosacontrolplane_controller.go:412]: What does RawID() return?

Comment (muraee) [reply controlplane/rosa/controllers/rosacontrolplane_controller.go:412]: it returns the cluster version in the form of `x.y.z` like `4.14.5`
ID() would return something like `openshift-v4.14.5` or possibly `openshift-v4.14.5-<channel>`

File: controlplane/rosa/api/v1beta2/rosacontrolplane_types.go (hunk 1 above)
Comment (muraee) [reply controlplane/rosa/api/v1beta2/rosacontrolplane_types.go]: added semver pattern validation. ref: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

**Pull request merged**